### PR TITLE
Docs restructuring

### PR DESCRIPTION
**Pull request by provinzkraut**
# DO NOT MERGE

This restructures our documentation and splits it into two parts:

1. General "Starlite website"
2. Documentation

The main purpose of this was to be able to only version the documentation and not the whole page.
The split achieves several things:

- Allows us to change and develop the main page in a way that viewing the docs for an old version won't also get you old content for non-documentation pages
- Changes to theme and such will be available in old versions of the docs as well
- Makes the versioning future proof, in that we can still opt to re-do the whole thing, and serve a snapshot of an old documentation from within it, while the old docs can also link to the newer version

Also adds a landing page, which is only temporary and aims to be "good enough" for now. @JacobCoffee will take care of creating something more visually appealing at some point.

Preview is available at https://provinzkraut.github.io/starlite/.

I'd like some feedback on the content of the landing page specifically, since I'm not entirely sure about its content and phrasing. 

**Todo before merge**

- [ ] Backport changes to the `v1.51` branch
- [ ] Manually do initial deploy to GH pages

### CODE REVIEW

File: docs/_templates/landing-page.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            <p>
+                Fully compliant with the ASGI specification, Starlite is asynchronous at heart, but synchronous
+                execution is not a second class citizen: Synchronous applications run without performance penalties
+            </p>
+        </section>
+
+        <section>
+            <h2>Opinionated yet flexible</h2>
+            <p>
+                Strongly encourages best practices, but tries to not get into the way of adapting it into whatever
```
Comment (Goldziher) [inline docs/_templates/landing-page.html:43]: -
```suggestion
                Strongly encourages best practices, but tries not to get in the way of adapting it into whatever
```

File: docs/_templates/landing-page.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            <p>
+                Bottom-up architecture makes it easy to develop modular and decoupled code, allowing applications to
+                scale easily, both in development and deployment
+            </p>
+        </section>
+
+        <section>
+            <h2>Community driven</h2>
+            <p>
+                Developed by a community of volunteers, and lead by a team of maintainers, Starlite is a community
```
Comment (Goldziher) [inline docs/_templates/landing-page.html:59]:
```suggestion
                Developed by a community of volunteers, and led by a team of maintainers, Starlite is a community
```

File: docs/about/organization.rst [hunk 3]
```diff
@@ -1,3 +1,5 @@
+:html_theme.sidebar_secondary.remove:
+
 Organization
```
Comment (Goldziher) [inline docs/about/organization.rst:3]: In general - there is a mix of English and British spellings in the docs. I suggest we keep consistent. In the above we have American, which i personally prefer. Let's stick with one form.

File: docs/lib/index.rst [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+    All extras:
+        :code:`pip install starlite[full]`
+
+
+
+Minimal Example
+---------------
+
+**Define your data model** using pydantic or any library based on it (for example ormar, beanie, SQLModel):
```
Comment (Goldziher) [inline docs/lib/index.rst:60]: We allow using dataclasses, typed dicts etc. I suggest we don't get bogged down with pydantic.

File: docs/lib/index.rst [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+    from my_app.models import User
+
+
+    class UserController(Controller):
+        path = "/users"
+
+        @post()
+        async def create_user(self, data: User) -> User:
+            ...
```
Comment (Goldziher) [inline docs/lib/index.rst:109]: One complaint that was often repeated is that the basic examples are not runnable due to the use of ellipses. 

I suggest remaking this example- simpler and runnable. I understand of course if it's our if scope.

File: docs/lib/index.rst [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+   app = Starlite(route_handlers=[UserController])
+
+To **run your application**, use an ASGI server such as `uvicorn <https://www.uvicorn.org/>`_ :
+
+.. code-block:: shell
+
+   uvicorn my_app.main:app --reload
+
+
+Philosophy
```
Comment (Goldziher) [inline docs/lib/index.rst:150]: I suggest this section is split into a different page and rewritten .

File: docs/lib/index.rst [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+----------
+
+- Starlite is a community-driven project. This means not a single author,
+  but rather a core team of maintainers is leading the project, supported by a community
+  of contributors. Starlite currently has 5 maintainers and is being very actively developed.
+- Starlite draws inspiration from `NestJS <https://nestjs.com/>`_ - a contemporary TypeScript framework - which places
```
Comment (Goldziher) [inline docs/lib/index.rst:156]: I would remove this completely. Inspiration is finished 😁

File: docs/lib/index.rst [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+  opinions and patterns at its core.
+- While still allowing for **function-based endpoints**, Starlite seeks to build on Python's powerful and versatile
+  OOP, by placing **class-based controllers** at its core.
+- Starlite is **not** a microframework. Unlike frameworks such as FastAPI, Starlette or Flask, Starlite includes a lot
```
Comment (Goldziher) [inline docs/lib/index.rst:160]: This is all to compressed

**Review by Goldziher — COMMENTED** [review]
Done. Can you attach some pictures to the PR? I don't have a visualisation in mind.

**Comment by provinzkraut** [comment]
> Can you attach some pictures to the PR?

Just check out the link, it's live there.

**Comment by provinzkraut** [comment]
>  In mobile i have to open a menu to switch to dark mode. Also, it doesn't switch to my device default theme automaticall.

I'll check if the new theme supports this. On desktop the navigation should be a bit easier, and to be honest, it's what I aimed for mostly. Mobile should work, but it's not the main target really.

**Comment by Goldziher** [comment]
> >  In mobile i have to open a menu to switch to dark mode. Also, it doesn't switch to my device default theme automaticall.
> 
> I'll check if the new theme supports this. On desktop the navigation should be a bit easier, and to be honest, it's what I aimed for mostly. Mobile should work, but it's not the main target really.

I strongly disagree. I read docs on mobile often, and so do many others.

**Comment by provinzkraut** [comment]
@peterschutt, @cofin, @JacobCoffee it'd be great if one of you native speakers could check on the wording an grammar and such of the landing page.

File: docs/about/organization.rst (hunk 3 above)
Comment (provinzkraut) [reply docs/about/organization.rst:3]: I agree, we should stick to one.

File: docs/lib/index.rst (hunk 4 above)
Comment (provinzkraut) [reply docs/lib/index.rst:60]: Agree, but I don't want to change docs content in this PR. This should be purely structural.

File: docs/lib/index.rst (hunk 6 above)
Comment (provinzkraut) [reply docs/lib/index.rst:150]: @Jarmos-san is working on that in #1153

**Comment by provinzkraut** [comment]
> I strongly disagree. I read docs on mobile often, and so do many others.

I'll see what I can do!

**Comment by provinzkraut** [comment]
@Goldziher Can you check the mobile header again?

**Comment by Goldziher** [comment]
> @Goldziher Can you check the mobile header again?

It's perfect!

File: docs/_templates/landing-page.html [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+                Fully compliant with the ASGI specification, Starlite is asynchronous at heart, but synchronous
+                execution is not a second class citizen: Synchronous applications run without performance penalties
+            </p>
+        </section>
+
+        <section>
+            <h2>Opinionated yet flexible</h2>
+            <p>
+                Strongly encourages best practices, but tries not to get in the way of adapting it into whatever
+                solution is needed
```
Comment (peterschutt) [inline docs/_templates/landing-page.html:49]: How about something like:

```suggestion
                Developed to encourage best practices, but flexible enough to stay out of your way
```

**Review by peterschutt — APPROVED** [review]

**Comment by JacobCoffee** [comment]
> @JacobCoffee it'd be great if one of you native speakers could check on the wording an grammar and such of the landing page.

will look. everything looks very good at a glance

File: docs/_templates/landing-page.html (hunk 9 above)
Comment (provinzkraut) [reply docs/_templates/landing-page.html:49]: That sounds better!

**Review by Jarmos-san — APPROVED** [review]

**Review by Goldziher — APPROVED** [review]

**Review by JacobCoffee — APPROVED** [review]

**Pull request merged**